core/domain/story_domain.py

```diff
@@ -1018,3 +1024,101 @@ def to_human_readable_dict(self):
             'title': self.title,
             'description': self.description
         }
+
+
+class StoryRights(object):
+    """Domain object for story rights."""
+
+    def __init__(self, story_id, manager_ids, story_is_published):
+        """Constructs a StoryRights domain object.
+
+        Args:
+            story_id: str. The id of the story.
+            manager_ids: list(str). The id of the users who have been assigned
+                as managers for the story.
+            story_is_published: bool. Whether the story is viewable by a
+                learner.
+        """
+        self.id = story_id
+        self.manager_ids = manager_ids
+        self.story_is_published = story_is_published
+
+    def to_dict(self):
+        """Returns a dict suitable for use by the frontend.
+
+        Returns:
+            dict. A dict version of StoryRights suitable for use by the
+                frontend.
+        """
+        return {
+            'story_id': self.id,
+            'manager_names': self.manager_ids,
+            'story_is_published': self.story_is_published
+        }
+
+    def is_manager(self, user_id):
+        """Checks whether given user is a manager of the story.
+
+        Args:
+            user_id: str or None. Id of the user.
+
+        Returns:
+            bool. Whether user is a story manager of this story.
+        """
+        return bool(user_id in self.manager_ids)
+
+
+class StoryRightsChange(object):
+    """Domain object for changes made to a story rights object."""
+
+    OPTIONAL_CMD_ATTRIBUTE_NAMES = [
+        'assignee_id', 'new_role', 'old_role'
+    ]
+
+    def __init__(self, change_dict):
+        """Initialize a StoryRightsChange object from a dict.
+
+        Args:
+            change_dict: dict. Represents a command. It should have a 'cmd'
+                key, and one or more other keys. The keys depend on what the
+                value for 'cmd' is. The possible values for 'cmd' are listed
+                below, together with the other keys in the dict:
+                - 'change_role' (with assignee_id, new_role and old_role)
+                - 'create_new'
+                - 'publish_story'
+                - 'unpublish_story'
+
+        Raises:
+            Exception: The given change dict is not valid.
+        """
+        if 'cmd' not in change_dict:
+            raise Exception('Invalid change_dict: %s' % change_dict)
+        self.cmd = change_dict['cmd']
+
+        if self.cmd == CMD_CHANGE_ROLE:
+            self.assignee_id = change_dict['assignee_id']
+            self.new_role = change_dict['new_role']
+            self.old_role = change_dict['old_role']
+        elif self.cmd == CMD_CREATE_NEW:
+            pass
+        elif self.cmd == CMD_PUBLISH_STORY:
+            pass
+        elif self.cmd == CMD_UNPUBLISH_STORY:
+            pass
+        else:
+            raise Exception('Invalid change_dict: %s' % change_dict)
+
+    def to_dict(self):
```

Comment:
Missing tests for to_dict() (and, for that matter, `__init__`())?

Reply:
Added.